### PR TITLE
fix(gui): preserve terminal snapshot boundary scrollback

### PR DESCRIPTION
**Pull request by akiojin**
## Summary

- Preserve the scrollback boundary line when reconnect snapshots replay terminal history.
- Add a regression test for the `\r\n` case where the visible grid has completed lines plus a blank cursor row.

## Changes

- `crates/gwt-terminal/src/pane.rs`: compute scrollback overlap from the cursor row instead of the raw terminal height.
- `crates/gwt-terminal/src/pane.rs`: add a focused RED/GREEN regression covering the missing `line-13` boundary case.

## Testing

- `cargo test -p gwt-terminal test_snapshot_bytes_preserves_boundary_scrollback_line --all-features -- --nocapture`
- `cargo test -p gwt-terminal --all-features`
- `cargo test -p gwt app_runtime_frontend_ready_replays_terminal_snapshot_with_scrollback_history --all-features`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `cargo fmt -- --check`
- `git diff --check`
- `bunx --bun markdownlint-cli . --config .markdownlint.json --ignore target --ignore CHANGELOG.md --ignore tasks/todo.md`
- `cargo build -p gwt --bin gwt --bin gwtd`
- `bunx commitlint --from origin/develop --to HEAD`

## PR Readiness

- State: Ready for review
- Releaseable slice: yes
- Known blockers: none
- Remaining acceptance: none
- User verification: SPEC-2008 scroll UX was confirmed on 2026-06-01; this follow-up is covered by the new automated boundary regression.

## Closing Issues

None

## Related Issues

- #2008
- Follow-up to #2965

## Checklist

- [x] Tests added or updated
- [x] Lint and format checks passed
- [x] User verification status documented
- [x] CHANGELOG impact considered: patch fix
- [x] Documentation update not required for this boundary-condition fix

### CODE REVIEW

File: crates/gwt-terminal/src/pane.rs [hunk 1]
```diff
@@ -145,8 +145,8 @@ impl Pane {
     pub fn snapshot_bytes(&self) -> Vec<u8> {
         let mut snapshot = Vec::new();
         let scrollback_len = self.scrollback.len();
-        let visible_rows = usize::from(self.parser.screen().size().0);
-        let replayable_len = scrollback_len.saturating_sub(visible_rows);
+        let visible_completed_lines = usize::from(self.parser.screen().cursor_position().0);
+        let replayable_len = scrollback_len.saturating_sub(visible_completed_lines);
```
[the remark this answers is not in the document]
Comment (akiojin) [reply crates/gwt-terminal/src/pane.rs:149]: Fixed in follow-up PR #2967 with commit `3b38c9263`: snapshot overlap now matches the current visible rows against the scrollback suffix instead of using the cursor row as a proxy. The new `test_snapshot_bytes_uses_visible_rows_for_scrollback_overlap` regression covers the CSI cursor-movement case where `line-14` was replayed and then drawn again.
